_posts/2025-03-31-zscore-hybrid-search.md

```diff
@@ -0,0 +1,170 @@
+---
+layout: post
+title:  Z Score Normalization Technique for Hybrid Search
+authors:
+  - kazabdu
+  - gaievski
+date: 2025-03-31
+has_science_table: true
+categories:
+  - technical-posts
+meta_keywords: z score normalization, OpenSearch 3.0-beta, neural search plugin, hybrid search, relevance ranking, search normalization, k-nn search, L2 normalization, how reciprocal rank fusion works
+meta_description: Learn about z score normalization using the Neural Search plugin in OpenSearch 3.0-beta. Discover how this new approach to hybrid search merges results from multiple query sources for improved relevance.
+---
+
+In the world of search engines and machine learning, data normalization plays a crucial role in ensuring fair and accurate comparisons between different features or scores. 
+Hybrid query uses multiple normalization techniques for preparing final results, main two types are score based normalization and rank base combination. In score base normalization, min-max normalization doesn’t work well with outliers (Outliers are data points that significantly differ from other observations in a dataset. 
```

Comment:
Can we mention that l2 is also an option for score based normalization and maybe a weakness of that one?

Reply:
l2 is not affected by outliers by that much as it covers relative magnitudes of features. And this blogpost is moreover comparison between with min max